File: LeetCode/2966/Solution.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        Arrays.sort(nums);

        int n = nums.length;
        int[][] result = new int[n/3][3];
        for(int i=0;i<n/3;++i) {
            for(int j=0;j<3;++j) {
                result[i][j] = nums[i*3+j];
            }
        }

        for(int i=0;i<n/3;++i) {
            if(result[i][1]-result[i][0] > k || result[i][2]-result[i][0] > k) {
                return new int[][]{};
            }
        }

        return result;
    }
}

File: LeetCode/2966/Solution.kt
class Solution {
    fun divideArray(nums: IntArray, k: Int): Array<IntArray> {
        nums.sort()
        for(i in 0 until nums.size / 3) {
            if((nums[3*i+2] - nums[3*i]) > k) {
                return arrayOf()
            }
        }
        val result = Array<IntArray>(nums.size / 3) { IntArray(3) }
        for(i in 0 until nums.size / 3) {
            val temp = IntArray(3)
            for(j in 0 until 3) {
                temp[j] = nums[i*3+j]
            }
            result[i] = temp
        }
        return result
    }
}
